libraries/AP_Compass/AP_Compass_RM3100.cpp

```diff
@@ -117,13 +117,15 @@ bool AP_Compass_RM3100::init()
         ccy1 != CCP1_DEFAULT || ccy0 != CCP0_DEFAULT ||
         ccz1 != CCP1_DEFAULT || ccz0 != CCP0_DEFAULT) {
         // couldn't read one of the cycle count registers or didn't recognize the default cycle count values
-        goto fail;
+        // hal.console->printf("RM3100: Unable to read the register for whoami test\n");
+        dev->get_semaphore()->give();
+        return false;
     }
 
     dev->setup_checked_registers(8);
 
-    dev->write_register(RM3100_TMRC_REG, TMRC, true); // cycle count z
-    dev->write_register(RM3100_CMM_REG, CMM, false); // CMM configuration
+    dev->write_register(RM3100_TMRC_REG, TMRC, true); // CMM data rate
```

Comment:
why is TMRC being set to true? It's a speed value. true will be 1, which is an invalid value from the CMM update rate table.

Reply:
I take that from the lastest patch of the author of this backend ..., but you are right, it is wrong.

Reply:
If I understand well the question the `true` in `dev->write_register(RM3100_TMRC_REG, TMRC, true)` is here for setting whether or not a check is performed on this register ? The value of the register is set to `TMRC=0x94`.

Reply:
yes, sorry, you're right